pkg/tsdb/prometheus/querydata/response.go

```diff
@@ -123,7 +123,7 @@ func addMetadataToMultiFrame(q *models.Query, frame *data.Frame, enableDataplane
 		frame.Fields[1].Config = &data.FieldConfig{DisplayNameFromDS: customName}
```

Comment:
I believe things should be taking the name from here `DisplayNameFromDS` before they take it from the frame name - @ryantxu thoughts on this?

If this is the case, the following `&& custonName` should not be needed.

Reply:
seriesToRows uses [getFrameDisplayName](https://github.com/grafana/grafana/blob/6072b7d5ff72e9d28ef125db5553859d42c9867f/packages/grafana-data/src/field/fieldState.ts#L7-L37) which returns `frame.name` first if it is there. The logic in this function never gets to the field name check which uses displayNameFromDS unfortunately.

Reply:
Okay, well still want to hear what Ryan thinks about the precedence in getFrameDisplayName. Seems odd to me that labels take precedence over the DisplayNameFromDS.

Reply:
This makes sense, see what @ryantxu thinks. 

The DisplayNameFromDS is only set on the field config though. When getting the Frame name, we used to override this with the custom name from the legend option. Now we set the custom name in the second field config like so:

`frame.Fields[1].Config = &data.FieldConfig{DisplayNameFromDS: customName}`

We could change the `getFrameDisplayName` to return this if dataplane is enabled but this seems a little odd. 

```
export function getFrameDisplayName(frame: DataFrame, index?: number) {
  if (prometheusDataplane && dataplaneFrontendFallback) {
    return frame.fields[1].config.displayNameFromDS
  }

  if (frame.name) {
    return frame.name;
  }
```